sources/osg/State.js

```diff
@@ -55,6 +55,9 @@ define( [
         // texture manager is referenced here because it's associated with gl object
         // of the gl context intialized with State
         this._textureManager = new TextureManager();
+
+
+        this._programCommonUniformsCache = {};
```

Comment:
Did you try using Map from es6? I don't know if it could improve anything, just wondering...